evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -94,8 +94,18 @@ def transform(self, X, y=None):
             pd.DataFrame, pd.Series: Transformed features and target.
         """
         X, y = self._prepare_data(X, y)
+
+        categorical_columns = X.ww.select("Categorical", return_schema=True).columns
+        if len(categorical_columns):
```

Comment:
Might be a bit of a nit, but can't we just iterate over `categorical_columns` without the if?  I assume the select returns an empty iterable vs. None.

Reply:
Great point!